storage/index/types.go

```diff
@@ -183,41 +178,111 @@ type BlockTickResult struct {
 	NumDocs     int64
 }
 
+// WriteBatchEntry captures a document to index, and the lifecycle hooks to call thereafter.
+type WriteBatchEntry struct {
```

Comment:
should retain the changes to this type

Reply:
might want to add on a ReceivedTime to capture e2e index latency with this struct too